### PR TITLE
[FIX] odoo: dump stacktrace of timed out workers

### PR DESCRIPTION
**Pull request by Icallhimtest**
So the logs contain some indication as to what was exceeding the limits.

multi-worker version.

multithread targetting v12.0 here: https://github.com/odoo/odoo/pull/36586

### CODE REVIEW

File: odoo/service/server.py [hunk 1]
```diff
@@ -742,11 +742,29 @@ def process_timeout(self):
...
+                    worker.stack_dumped = now
+                elif worker.stack_dumped and worker.stack_dumped > now - self.beat:
+                    pass
+                else:
+                    _logger.info("Killing %s (%s)",
+                                 worker.__class__.__name__,
+                                 pid)
+                    self.worker_kill(pid, signal.SIGKILL)
+            elif worker.stack_dumped:
+                _logger.info("%s (%s) just finished, keeping it alive",
```
Comment (KangOl) [inline odoo/service/server.py]: Is it true? `SIGQUIT` has already been sent.

Comment (Icallhimtest) [reply odoo/service/server.py]: `SIGQUIT` only does a dumpstacks, I don't see a problem to keep the worker alive afterwards, do you?

Note that this cornercase is very unlikely to happen, the request would need to last exactly limit_time_real + [0-8] seconds.

**Comment by Icallhimtest** [comment]
@KangOl seems like we've missed doing it in saas-12.5

I've retargeted it for 13.0, did you still have a problem with the way this is implemented?

@odony Okay to add this to v13.0 now?

I've "subtly" renamed this from "IMP" to "FIX" as it could be considered as a missing element from https://github.com/odoo/odoo/commit/bd3e510edbcd62f508dae28f573ee4d1a51bcf87 ...

I'm not sure how you value this PR. Personally I like it, and know it'll come in handy from time to time even if on other occasions it may pollute the logs a bit.

**Comment by lathama** [comment]
@Icallhimtest knowing that 13 is no longer updated, is this PR still needed?

**Comment by Icallhimtest** [comment]
> @Icallhimtest knowing that 13 is no longer updated, is this PR still needed?

I think it would be a nice QoL improvement for people on-premise/SaaS in multi-worker mode allowing to debug timeouts in the logs same as they can do on Odoo.sh. But yeah, it's more of a job for the python framework team to decide if they want it or not.

I'll close this one as it needs to be remade for a more recent version.

**Comment by Julien00859** [comment]
@Icallhimtest don't wait for @odoo/rd-framework-py to take action regarding the management of the servers, you guys from SH/Online are much knowledgeable than we are :p

**Pull request closed without merging**